src/v/iceberg/tests/compatibility_test.cc

```diff
@@ -325,3 +325,613 @@ struct_type nested_test_struct() {
     return nested_struct;
 }
 } // namespace
+
+struct struct_evolution_test_case {
+    std::string_view description{};
+    std::function<struct_type(unique_id_generator&)> generator;
+    std::function<void(struct_type&)> update;
+    checked<std::nullopt_t, schema_evolution_errc> err{std::nullopt};
+    std::function<bool(const struct_type&, const struct_type&)> validator =
+      [](const struct_type&, const struct_type&) { return true; };
+    schema_changed any_change{true};
+};
+
+std::ostream&
+operator<<(std::ostream& os, const struct_evolution_test_case& tc) {
+    return os << tc.description;
+}
+
+static const std::vector<struct_evolution_test_case> valid_cases{
```

Comment:
Wondering whether the current code supports multiple updates in the same evolution?

Reply:
you mean prior to this diff? I think so

Reply:
I meant with the new compatibility checks -- for instance, I was wondering if it made sense to test an add and remove in the same evolution (unless i missed cases where multiple operations are happening)

Reply:
oh, yeah definitely. i don't think there's a specific test geared towards that, but the one whose description begins "grouping multiple fields into a struct..." has that effect